### PR TITLE
Add new options for the 'mpc status' command

### PR DESCRIPTION
**Pull request by ericyape**
Hi,

I'd like to share new options I have added to the 'mpc status' command:

1/ mpc status '%state%' now returns 'stopped' if the MPD server is stopped

2/ mpc status '%kbitrate% %audioformat% %audiochannels% %audiobits% %audiosamplerate%' now returns full details on the audio streams (or 'n/a' if not available)

Thanks in advance for considering these (small) contributions.

Eric

### CODE REVIEW

File: src/status_format.c [hunk 1]
```diff
@@ -75,8 +75,12 @@ status_value(const struct mpd_status *status, const char *name)
 	} else if (strcmp(name, "state") == 0) {
 		if (mpd_status_get_state(status) == MPD_STATE_PLAY) {
 			return "playing";
-		} else {
+		} else if (mpd_status_get_state(status) == MPD_STATE_PAUSE) {
 			return "paused";
+		} else if (mpd_status_get_state(status) == MPD_STATE_STOP) {
+			return "stopped";
+		} else {
+			return "n/a";
```
Comment (MaxKellermann) [inline src/status_format.c:83]: Should return `NULL` if there is no value.

File: src/status_format.c [hunk 2]
```diff
@@ -110,6 +110,36 @@ status_value(const struct mpd_status *status, const char *name)
 		} else {
 		    return "off";
 		}
+	} else if (strcmp(name, "kbitrate") == 0) {
+		sprintf(buffer, "%d", mpd_status_get_kbit_rate(status));
```
Comment (MaxKellermann) [inline src/status_format.c:114]: Bad formatter: the value is not signed.

File: src/status_format.c [hunk 3]
```diff
@@ -110,6 +110,36 @@ status_value(const struct mpd_status *status, const char *name)
...
+	} else if (strcmp(name, "audioformat") == 0) {
+		const struct mpd_audio_format *af = mpd_status_get_audio_format(status);
+		if (af) {
+			snprintf(buffer, sizeof(buffer), "%uHz/%ubits/%u", af->sample_rate, af->bits, af->channels);
+		} else {
+			return "n/a";
```
Comment (MaxKellermann) [inline src/status_format.c:120]: Bad return value

File: src/status_format.c [hunk 4]
```diff
@@ -110,6 +110,36 @@ status_value(const struct mpd_status *status, const char *name)
 		} else {
 		    return "off";
 		}
+	} else if (strcmp(name, "kbitrate") == 0) {
+		sprintf(buffer, "%d", mpd_status_get_kbit_rate(status));
+	} else if (strcmp(name, "audioformat") == 0) {
+		const struct mpd_audio_format *af = mpd_status_get_audio_format(status);
+		if (af) {
+			snprintf(buffer, sizeof(buffer), "%uHz/%ubits/%u", af->sample_rate, af->bits, af->channels);
```
Comment (MaxKellermann) [inline src/status_format.c:118]: This is different from the format used by MPD, see https://mpd.readthedocs.io/en/latest/user.html#global-audio-format
Having a different format in MPD and mpc is confusing.
Also your code doesn't consider floating point and DSD.

File: src/status_format.c [hunk 5]
```diff
@@ -110,6 +110,36 @@ status_value(const struct mpd_status *status, const char *name)
...
+		} else {
+			return "n/a";
+		}
+	} else if (strcmp(name, "audiosamplerate") == 0) {
+		const struct mpd_audio_format *af = mpd_status_get_audio_format(status);
+		if (af) {
+			sprintf(buffer, "%u", af->sample_rate);
+		} else {
+			return "n/a";
```
Comment (MaxKellermann) [inline src/status_format.c:127]: Bad return value

File: src/status_format.c [hunk 6]
```diff
@@ -110,6 +110,36 @@ status_value(const struct mpd_status *status, const char *name)
...
+		}
+	} else if (strcmp(name, "audiobits") == 0) {
+		const struct mpd_audio_format *af = mpd_status_get_audio_format(status);
+		if (af) {
+			sprintf(buffer, "%u", af->bits);
```
Comment (MaxKellermann) [inline src/status_format.c:132]: What about float and DSD?
Is `audiobits` really a good name for this?

File: src/status_format.c [hunk 7]
```diff
@@ -110,6 +110,36 @@ status_value(const struct mpd_status *status, const char *name)
...
+		} else {
+			return "n/a";
```
Comment (MaxKellermann) [inline src/status_format.c:134]: bad return value

File: src/status_format.c [hunk 8]
```diff
@@ -110,6 +110,36 @@ status_value(const struct mpd_status *status, const char *name)
...
+		}
+	} else if (strcmp(name, "audiochannels") == 0) {
+		const struct mpd_audio_format *af = mpd_status_get_audio_format(status);
+		if (af) {
+			sprintf(buffer, "%u", af->channels);
+		} else {
+			return "n/a";
```
Comment (MaxKellermann) [inline src/status_format.c:141]: Bad return value

**Review by MaxKellermann — CHANGES_REQUESTED** [review]
Documentation and NEWS line is missing

**Comment by ericyape** [comment]
Hi Max,
many thanks for the comments. I have tried to take into account all of them in the proposed commits.
Eric
PS : also change "audiobits" option to "bits" to stick to MPD documentation.

**Pull request merged**